### DISCUSSION TITLE
Open link in current tab vs a new tab

### DISCUSSION THREAD

**Discussion by njpig** [mozilla-mobile/firefox-ios#10183] [Q&A] [answered]
When a link is opened in a new tab, webView.load() is called to get the web content.

However, when a link is opened in current tab, I did not see webView.load() is called. Is webView.load() never called in this case? How does Firefox get the web content is this case then?

**Comment by lmarceau** [comment, marked as answer]
@njpig Click insides the webview are handled by WKWebView. There's a delegate called [WKNavigationDelegate](https://developer.apple.com/documentation/webkit/wknavigationdelegate) that provides methods to know when a link is opened in a web view, if it should be navigated to.

**Reply by njpig** [reply]
> WKNavigationDelegate

 @lmarceau I would like to change some headers in the http request before the request is sent. However, looks like the web contents have been received before any method of WKNavigationDelegate is called. Therefore, I have no way to change the headers if a link is opened in the current tab.

Do you have any comments?

**Reply by njpig** [reply]
@lmarceau Thanks a lot! 
It worked after setValue() and webView.load() was called in: 
func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction, decisionHandler: @escaping (WKNavigationActionPolicy) -> Void)

BTW, I had not known the work flow of the webView.
